pyproject.toml

```diff
@@ -26,10 +26,12 @@ keywords = ["aiida", "workflows"]
 requires-python = ">=3.8"
 dependencies = [
   "py~=1.11",
-  "pydantic~=1.10,>=1.10.8",
+  "pydantic~=2.7",
+  "pydantic-settings~=2.2",
   "python-dotenv~=1.0",
   "typer[all]~=0.9",
   "pyyaml~=6.0",
+  "eval_type_backport",
```

Comment:
Hmm, been a while, but I think I added this because of https://github.com/pydantic/pydantic/issues/7873. Not sure if it's still necessary, let me check.